sys/net/gnrc/sock/tcp/gnrc_sock_tcp.c

```diff
@@ -0,0 +1,146 @@
+    return gnrc_tcp_open(sock, remote, local_port);
+}
+
+int sock_tcp_listen(sock_tcp_queue_t *queue, const sock_tcp_ep_t *local,
+                    sock_tcp_t *queue_array, unsigned queue_len, uint16_t flags)
+{
+    /* Asserts defined by API. */
+    assert(queue != NULL);
+    assert(local != NULL);
+    assert(local->port != 0);
+    assert(queue_array != NULL);
+    assert(queue_len != 0);
+
+    /* Asserts to protect GNRC_TCP. Flags are not supported. */
+    assert(flags == 0);
+
+    /* Initialize given TCB queue, all given tcbs and forward call */
+    gnrc_tcp_tcb_queue_init(queue);
+    for (unsigned i = 0; i < queue_len; ++i) {
+        gnrc_tcp_tcb_init(&queue_array[i]);
+    }
+    return gnrc_tcp_listen(queue, queue_array, queue_len, local);
+}
+
+void sock_tcp_disconnect(sock_tcp_t *sock)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    gnrc_tcp_close(sock);
+}
+
+void sock_tcp_stop_listen(sock_tcp_queue_t *queue)
+{
+    /* Asserts defined by API. */
+    assert(queue != NULL);
+    gnrc_tcp_stop_listen(queue);
+}
+
+int sock_tcp_get_local(sock_tcp_t *sock, sock_tcp_ep_t *ep)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    assert(ep != NULL);
+    return gnrc_tcp_get_local(sock, ep);
+}
+
+int sock_tcp_get_remote(sock_tcp_t *sock, sock_tcp_ep_t *ep)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    assert(ep != NULL);
+    return gnrc_tcp_get_remote(sock, ep);
+}
+
+int sock_tcp_queue_get_local(sock_tcp_queue_t *queue, sock_tcp_ep_t *ep)
+{
+    /* Asserts defined by API. */
+    assert(queue != NULL);
+    assert(ep != NULL);
+    return gnrc_tcp_queue_get_local(queue, ep);
+}
+
+int sock_tcp_accept(sock_tcp_queue_t *queue, sock_tcp_t **sock, uint32_t timeout)
+{
+    /* Asserts defined by API. */
+    assert(queue != NULL);
+    assert(sock != NULL);
+
+    /* Map SOCK_NO_TIMEOUT to 0. 0 is used in GNRC_TCP for no timeout */
+    if (timeout == SOCK_NO_TIMEOUT) {
+        timeout = 0;
+    }
+
+    /* Forward call to gnrc_tcp_accept.
+     * NOTE: Errorcodes -ECONNABORTED, -EPERM are not returned by
+     * gnrc_tcp_accept. All other error codes share the same semantics. */
+    return gnrc_tcp_accept(queue, sock, timeout);
+}
+
+ssize_t sock_tcp_read(sock_tcp_t *sock, void *data, size_t max_len, uint32_t timeout)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    assert(data != NULL);
+    assert(max_len > 0);
+
+    /* Map SOCK_NO_TIMEOUT to 0. 0 is used in GNRC_TCP for no timeout */
+    if (timeout == SOCK_NO_TIMEOUT) {
+        timeout = 0;
+    }
+
+    /* Forward call to gnrc_tcp_recv: All error codes share the same semantics */
+    return gnrc_tcp_recv(sock, data, max_len, timeout);
+}
+
+ssize_t sock_tcp_write(sock_tcp_t *sock, const void *data, size_t len)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    assert(data != NULL);
+    assert(len > 0);
```

Comment:
calling this with `len = 0` should not be an error.
If `gnrc_tcp_send()` can't handle this we should at least 

```suggestion
    if (len == 0) {
        return 0;
    }
```

Reply:
> The entire use case is nonsense and i am not sure is gnrc_tcp handles it correctly.

I can easily imagine a loop like

```C
int res;
while ((res = gnrc_tcp_recv(tcb, data, data_len, 0) {
    data += res;
    data_len -= res;
}
```
That shouldn't trigger an assertion when `data_len` reaches 0. Same for `gnrc_tcp_send()`.

The API also does not specify that the number of bytes to those functions must not be zero.

On Linux I can `read()` 0 bytes without causing a kernel panic :wink:

Reply:
I'll fix it after this is merged #16738